File: 课程网站/microlectures/2.4-stability/source/first_experiment.m
% Standalone; zero-state convolution. Finite plots are not convergence proofs.
n=0:127; hs={0.8.^n,1.2.^n,1./(n+1)}; names={'a=0.8','a=1.2','harmonic'};
f=figure('Visible','off');
for j=1:3
 h=hs{j}; y=conv(h,ones(size(n))); y=y(1:128);
 assert(max(abs(y-cumsum(h))./max(1,abs(y)))<1e-12);
 subplot(2,3,j); stem(n,h,'.'); title([names{j} ' h[n]']);
 subplot(2,3,j+3); plot(n,y); title('unit-step output'); xlabel('n');
end
exportgraphics(f,fullfile(fileparts(mfilename('fullpath')),'first-experiment-matlab.png')); close(f);
disp('MATLAB standalone experiment passed');

File: 课程网站/microlectures/2.4-stability/source/first_experiment.py
"""Standalone: Python + numpy + matplotlib. Finite curves are not infinite-sum proofs."""
from pathlib import Path
import numpy as np
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt
n=np.arange(128);fig,axes=plt.subplots(2,3,figsize=(12,6))
for j,(name,h) in enumerate([('a=0.8',.8**n),('a=1.2',1.2**n),('harmonic',1/(n+1))]):
 y=np.convolve(h,np.ones(128))[:128]
 assert np.allclose(y,np.cumsum(h))
 axes[0,j].stem(n,h,markerfmt='.');axes[0,j].set_title(name+' h[n]')
 axes[1,j].plot(n,y);axes[1,j].set_title('unit-step output');axes[1,j].set_xlabel('n')
fig.tight_layout();fig.savefig(Path(__file__).with_name('first-experiment-python.png'));print('Python experiment passed')
